src/service.py

```diff
@@ -49,15 +51,18 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         self.snap_client.set(snap_config, typed=True)
 
 
-def snap_install(resource: str) -> SnapService:
-    """Install or refresh the snap, and return the snap service.
+def snap_install() -> SnapService:
+    """Install the snap, and return the snap service.
+
+    Before installing the snap, it will try to remove the upstream snap that could be installed on
+    older versions of this charm.
 
     Raises an exception on error.
     """
+    remove_upstream_snap()
     try:
-        logger.debug("installing snap.")
-        logger.debug("fetching from resource.")
-        snap_client = snap.install_local(resource, dangerous=True)
+        snap_client = snap.add(SNAP_NAME, channel="latest/stable")
```

Comment:
hardcoding a channel name is generally not a good idea since it will prevent testing with latest/edge or latest/candidate for example.